lib/service/functions.js

```diff
@@ -40,18 +40,20 @@ function process(task, modules, callback) {
     return callback(task);
   }
 
-  const context = {};
-  const currentContext = task.hookType === 'post' ? task.response : task.request;
+  const { method, username, userId, entityId, loginOptions } = task.request;
+  const context = { method, username, userId, entityId, loginOptions };
+
+  for (const key in context) {
+    if (context[key] == undefined) delete context[key];
+  }
```

Comment:
Well, this looks a bit odd. A simpler logic would be to iterate over a list of keys and assign them to an empty context object if they exist. Something like:
```
const context = {};
for (const key in ['method', 'username', 'userId', 'entityId', 'loginOptions']) {
  if (task.request[key] != null) {
    context[key] = task.request[key];
  }
}
```

Reply:
Done